### ISSUE TITLE
boss_migrate cannot handle "schema_migrations"?

### ISSUE THREAD

**Issue by liuzhen** [ErlyORM/boss_db#198]
Hi, I was working with boss_db, when I found the db migration is left out from ChicagoBoss (CB), then I also add CB into rebar deps.

What I have done:

1 boss_migrate:make()
2 boss_migrate:run()

In the same erl, boss_db:execute() works well.
# What did I do wrong?

boss_migrate:run(app_name_atom).  
18:36:22.724 [info] Reading migration file: "/_Path_/priv/migrations/1413279116_create_table.erl"
*\* exception exit: {{undef,
                        [{schema_migrations,module_info,[exports],[]},
                         {boss_record_lib,dummy_record,1,
                             [{file,"src/boss_record_lib.erl"},{line,75}]},
                         {boss_record_lib,database_table,1,
                             [{file,"src/boss_record_lib.erl"},{line,99}]},
                         {boss_db_adapter_mysql,table_exists,2,
                             [{file,
                                  "src/db_adapters/boss_db_adapter_mysql.erl"},
                              {line,105}]},
                         {boss_db_controller,handle_call,3,
                             [{file,"src/boss_db_controller.erl"},{line,192}]},
                         {gen_server,handle_msg,5,
                             [{file,"gen_server.erl"},{line,580}]},
                         {proc_lib,init_p_do_apply,3,
                             [{file,"proc_lib.erl"},{line,239}]}]},

**Comment by Bunlong** [comment]
My problem with Mysql:

> boss_migrate:run(address).
09:55:52.987 [info] Reading migration file: "/home/bunlong/workspace/my_project/geekhmerlab/chicagoboss/addressbook/priv/migrations/1420682718_address.erl"
** exception exit: {{undef,[{boss_db_adapter_mock,table_exists,
                                                  [undefined,schema_migrations],
                                                  []},
                            {boss_db_controller,handle_call,3,
                                                [{file,"src/boss_db_controller.erl"},{line,188}]},
                            {gen_server,try_handle_call,4,
                                        [{file,"gen_server.erl"},{line,607}]},
                            {gen_server,handle_msg,5,
                                        [{file,"gen_server.erl"},{line,639}]},
                            {proc_lib,init_p_do_apply,3,
                                      [{file,"proc_lib.erl"},{line,237}]}]},
                    {gen_server,call,
                                [<0.293.0>,{table_exists,schema_migrations},30000]}}
     in function  gen_server:call/3 (gen_server.erl, line 190)
     in call from boss_pool:call/3 (src/boss_pool.erl, line 12)
     in call from boss_db:create_migration_table_if_needed/0 (src/boss_db.erl, line 122)
     in call from boss_db:migrate/1 (src/boss_db.erl, line 108)
(addressbook@bunlong)9> 09:55:52.989 [error] gen_server <0.293.0> terminated with reason: call to undefined function boss_db_adapter_mock:table_exists/2 from boss_db_controller:handle_call/3 line 188
09:55:52.989 [error] CRASH REPORT Process <0.293.0> with 2 neighbours exited with reason: call to undefined function boss_db_adapter_mock:table_exists(undefined, schema_migrations) in gen_server:terminate/7 line 804
09:55:52.990 [error] Supervisor {<0.100.0>,poolboy_sup} had child boss_db_controller started with {boss_db_controller,start_link,undefined} at <0.293.0> exit with reason call to undefined function boss_db_adapter_mock:table_exists(undefined, schema_migrations) in context child_terminated



Please help me to fix it. Thank you!

**Comment by davidw** [comment]
> boss_db_adapter_mock

Offhand, that doesn't look like Mysql...

**Comment by Bunlong** [comment]
@davidw: So how can I fix it Sir?

**Comment by davidw** [comment]
You should probably configure CB to use Mysql.  Or, even better, use Postgres :-)

**Comment by Bunlong** [comment]
@davidw: Yes I've already configured CB to use Mysql. I can connect to Mysql but I can run migrate. :(

**Comment by liuzhen** [comment]
hi
try manually creating the table of "schema_migrations" with the column definitions you may find in one boss_db src file.

**Comment by liuzhen** [comment]
@Bunlong 
"schema_migrations": id, version (varchar), migrated_at (like timestamp).

**Comment by peterk** [comment]
I have the same error when using mysql. Setting up the schema_migrations table manually with like this does not help.

```sql
CREATE TABLE `schema_migrations` (
  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
  `version` text NOT NULL,
  `migrated_at` datetime DEFAULT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```

**Comment by liuzhen** [comment]
create model for it also

发自我的 iPhone

**Comment by davidw** [comment]
A model is not necessary

**Comment by davidw** [comment]
Indeed, you should not create a model for that table.

Someone is going to need to hack on the Mysql adapter and fix things up if it is not currently functional.  The Postgres adapter is the one I originally added transactions to, so it's the one to use as a model.

Or, you could always just use Postgres :-)

**Comment by skalee** [comment]
It is already fixed in aforementioned #205.
